staging/src/k8s.io/client-go/tools/remotecommand/remotecommand.go

```diff
@@ -68,6 +68,29 @@ type streamExecutor struct {
 	protocols []string
 }
 
+// upgraderWithTerminateFunc wraps an upgrader with a terminate func.
+type upgraderWithTerminateFunc struct {
+	upgrader spdy.Upgrader
+	conns []httpstream.Connection
+}
+
+func (upgrader *upgraderWithTerminateFunc) NewConnection(resp *http.Response) (httpstream.Connection, error) {
+	conn, err := upgrader.upgrader.NewConnection(resp)
+	if err != nil {
+		return nil, err
+	}
+	upgrader.conns = append(upgrader.conns, conn)
```

Comment:
New connections may be created while/after `terminate()` is called. Please consider using a lock to protect it.

Reply:
OK, I will fix it.